### PR TITLE
Feature/fix forecast indexing and timeseries coords

### PR DESCRIPTION
**Pull request by rparker-creare**
HOTFIX: handling of forecast indexing and timeseries coordinates

### CODE REVIEW

File: podpac/core/coordinates/cfunctions.py [hunk 1]
```diff
@@ -25,14 +25,22 @@ def crange(start, stop, step, name=None):
         Signed, non-zero step between coordinates.
     name : str, optional
         Dimension name.
+    fix_stop_val : bool, optional
+        Default is True. If True, the stop value *may* be modified to ensure that
```
Comment (mpu-creare) [inline podpac/core/coordinates/cfunctions.py]: Looks like the default is False to be.

File: podpac/core/interpolation/nearest_neighbor_interpolator.py [hunk 2]
```diff
@@ -225,14 +227,20 @@ def _time_to_float(self, time, time_source, time_request):
...
+
         dtype0 = time_source.coordinates[0].dtype
         dtype1 = time_request.coordinates[0].dtype
         dtype = dtype0 if dtype0 > dtype1 else dtype1
         time = make_coord_value(time)
+
         if isinstance(time, np.datetime64):
             time = time.astype(dtype).astype(float)
+        elif isinstance(time, np.timedelta64):
+            time = time / np.timedelta64(1, 'ns')  # Convert timedelta64 to seconds as float
```
Comment (mpu-creare) [inline podpac/core/interpolation/nearest_neighbor_interpolator.py:239]: This gives floats not ints? 

Does this still work with the bounds are in units of second, or days?

Comment (mpu-creare) [reply podpac/core/interpolation/nearest_neighbor_interpolator.py:239]: I bet this won't work with _higher_precision_time_stack... well, that's my guess... I don't know how these interact off the top of my head.

File: podpac/core/coordinates/uniform_coordinates1d.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ class UniformCoordinates1d(Coordinates1d):
     step = tl.Union([tl.Float(), tl.Instance(np.timedelta64)], read_only=True)
     step.__doc__ = ":float, timedelta64: Signed, non-zero step between coordinates."
 
-    def __init__(self, start, stop, step=None, size=None, name=None, fix_stop_val=False):
+    def __init__(self, start, stop, step=None, size=None, name=None, fix_stop_val=True, fix_start_val=False):
```
Comment (mpu-creare) [inline podpac/core/coordinates/uniform_coordinates1d.py]: I think we should change it to `fix_end_val` and it can take values `None`, `start` or `stop`. Otherwise, you get in the situation where you set both `fix_start`/`fix_stop` to True, and then you end up with a non-uniform coordinates that have an inconsistent stop/start. If `None` it looks like the `step` is actually modified to make things work.

File: podpac/core/coordinates/uniform_coordinates1d.py [hunk 4]
```diff
@@ -129,9 +133,12 @@ def __init__(self, start, stop, step=None, size=None, name=None, fix_stop_val=Fa
         self.set_trait("stop", stop)
         self.set_trait("step", step)
 
-        if not fix_stop_val:  # Need to make sure that 'stop' is consistent with self.coordinates[-1]
```
Comment (mpu-creare) [inline podpac/core/coordinates/uniform_coordinates1d.py:132]: The original flag seemed to be `fix_step_val` instead of `fix_stop_val` to be consistent.

**Review by mpu-creare — CHANGES_REQUESTED** [review]
We need unit tests that fail under the previous implementation, but will now pass. The unit tests need to reference this issue, and the bug needs to be clearly explained. 

Also, please check the code for `datetime64` instances where maybe we should have a fix for `timedetla64` as well. You can start by flagging them with a comment and I can double-check if changes are needed.

File: podpac/core/coordinates/uniform_coordinates1d.py (hunk 3 above)
Comment (mswanson333) [reply podpac/core/coordinates/uniform_coordinates1d.py]: That's a better idea. I was intending that only one would be set to true at a time

Comment (mswanson333) [reply podpac/core/coordinates/uniform_coordinates1d.py]: I propose the variable should be called fix_boundary_val

File: podpac/core/interpolation/nearest_neighbor_interpolator.py (hunk 2 above)
Comment (mswanson333) [reply podpac/core/interpolation/nearest_neighbor_interpolator.py:239]: @mpu-creare I checked with seconds and days and it returns a float. What is a _higher_precision_time_stack?

File: podpac/core/interpolation/nearest_neighbor_interpolator.py [hunk 5]
```diff
@@ -87,18 +87,20 @@ def is_stacked(d):
...
+                    # Check if the first element is datetime64 or timedelta64
+                    if isinstance(bound_values[0], (np.datetime64, np.timedelta64)):
+                        # find the values to convert and save them
+                        reference_time = (
+                            eval_coordinates[dim] if dim in eval_coordinates.udims else source_coordinates[dim]
+                        )
+                        # convert found datetime64 or timedelta64 values to floats
+                        bounds[dim] = [
+                            self._atime_to_float(b, source_coordinates[dim], reference_time)
+                            for b in bound_values
```
Comment (mswanson333) [inline podpac/core/interpolation/nearest_neighbor_interpolator.py:102]: I'm going to make an issue to create a unit test for this after the release @mpu-creare @rparker-creare

**Comment by mswanson333** [comment]
@mpu-creare can you rereview these changes? I added tests for the nearest neighbor interpolator

File: podpac/core/coordinates/cfunctions.py [hunk 6]
```diff
@@ -25,14 +25,36 @@ def crange(start, stop, step, name=None):
...
         Dimension name.
+    anchor_boundary : str, optional
+        Determines whether the `start` or `stop` will be anchored while the other value 
+        may be adjusted to ensure consistency with the given `step` and `size`. 
+        Acceptable values are:
+        - `None`: the constructor will modify the `step` to be consistent with 
+            the `start` and `stop` boundaries defined by the user if necessary.
+        - `"start"` (default): The `start` value will be anchored while the `stop` value *may* be modified to ensure that:
+            ```
+            stop =
```
Comment (mpu-creare) [inline podpac/core/coordinates/cfunctions.py]: Something went wrong with the docstring here.

File: podpac/core/coordinates/test/test_uniform_coordinates1d.py [hunk 7]
```diff
@@ -1293,10 +1316,10 @@ def test_coordinates_floating_point_consistency(self):
         start = u.coordinates[0]
         stop = u.coordinates[-1]
         u2 = podpac.coordinates.UniformCoordinates1d(
-            start - 2 * step, stop + 2 * step + 1e-14, step=step, fix_stop_val=True
+            start - 2 * step, stop + 2 * step + 1e-14, step=step, anchor_boundary=None
         )
         u3 = podpac.coordinates.UniformCoordinates1d(
-            start - 2 * step, stop + 2 * step + 1e-14, size=u.size + 4, fix_stop_val=True
+            start - 2 * step, stop + 2 * step + 1e-14, size=u.size + 4, anchor_boundary=None
```
Comment (mpu-creare) [inline podpac/core/coordinates/test/test_uniform_coordinates1d.py:1322]: I don't see any examples of `anchor_bounary="start" or "stop"`. 

We really need to add tests for all three conditions. Also, I don't quite get why the asserts are for u2.start == u3.start? For `anchor_boundary=None`, should we be asserting on the steps?

**Review by mpu-creare — APPROVED** [review]
Glad to see some test. They still need work. I like the issue you have for the one code block. Please also make a note to test the different `anchor_boundary` values. 

I'm approving mostly because I know you're under a lot of time pressure.

File: podpac/core/coordinates/test/test_uniform_coordinates1d.py [hunk 8]
```diff
@@ -285,6 +285,29 @@ def test_datetime_year_step(self):
...
+    def test_anchor_boundaries(self):
+        # ascending, exact
+        c = UniformCoordinates1d("2025-01-01T12", "2025-01-03T00", "1,D", anchor_boundary=None)
+        # a = np.array(["2025-01-01T12", "2025-01-03T00"]).astype(np.datetime64)
+        assert c.start == np.datetime64("2025-01-01T12")
+        assert c.stop == np.datetime64("2025-01-03T00")
+        assert c.is_monotonic == True
+        assert c.is_descending == False
+        assert c.is_uniform == True
+        c = UniformCoordinates1d("2025-01-01T12", "2025-01-03T00", "1,D", anchor_boundary='start')
```
Comment (mswanson333) [inline podpac/core/coordinates/test/test_uniform_coordinates1d.py:297]: @mpu-creare the anchor_boundary values are tested here

File: podpac/core/coordinates/test/test_uniform_coordinates1d.py (hunk 7 above)
Comment (mswanson333) [reply podpac/core/coordinates/test/test_uniform_coordinates1d.py:1322]: There are tests for `anchor_bounary="start" or "stop"` see the comment in the code above

File: podpac/core/interpolation/nearest_neighbor_interpolator.py (hunk 5 above)
Comment (mswanson333) [reply podpac/core/interpolation/nearest_neighbor_interpolator.py:102]: The tests for these code changes are found in test_interpolators.py

File: podpac/core/coordinates/test/test_uniform_coordinates1d.py (hunk 8 above)
Comment (mpu-creare) [reply podpac/core/coordinates/test/test_uniform_coordinates1d.py:297]: Great job!  Not sure how I missed that.
